pod/perldelta.pod

```diff
@@ -5,35 +5,216 @@
+
+=head2 Switch and Smart Match operator reinstated
+
+The "switch" feature and the smartmatch operator, C<~~>, were introduced in
+v5.10.  Their behavior was significantly changed in v5.10.1.  When the
+"experiment" system was added in v5.18.0, switch and smartmatch were
+retroactively declared experimental.  Over the years, proposals to fix or
+supplement the features have come and gone.
+
+They were deprecated in Perl v5.38.0 and scheduled for removal in Perl
+5.42.0, and entirely removed in Perl 5.41.3.
+
+After some discussion these have been re-instated in 5.41.9.
+
+Using them no longer produces a deprecation warning.
+
+Switch itself still requires the C<switch> feature, which is enabled
+by default for feature bundles from v5.9.5 through to v5.34.  Switch
+remains disabled in feature bundles 5.35 and later, but can be
+separately enabled:
+
+  # no switch here
+  use v5.10;
+  # switch here
+  use v5.36;
+  # no switch here
+  use feature "switch";
+  # switch here
+
+Smart match now requires the C<smartmatch> feature, which is enabled
+by default and included in all feature bundles up to 5.40.  It is
+disabled for the 5.41 feature bundle and later, but can be separately
+enabled:
+
+  # smartmatch here
+  use v5.41;
+  # no smartmatch here
+  use feature "smartmatch";
+  # smartmatch here
+
+[L<GH #22752|https://github.com/Perl/perl5/issues/22752>]
+
+=head2 New C<SvVSTRING> API macro
+
+A new API macro has been added, which is used to obtain the second string
+buffer out of a "vstring" SV, in a manner similar to the C<SvPV> macro which
+obtains the regular string buffer out of a regular SV.
+
+    STRLEN len;
+    const char *vstr_pv = SvVSTRING(sv, vstr_len);
+
+See L<perlapi/C<SvVSTRING>>.
+
+=head2 Unicode 16.0 supported
+
+Perl now supports Unicode 16.0
+L<https://www.unicode.org/versions/Unicode16.0.0/> including the changes
+introduced in 15.1 L<https://www.unicode.org/versions/Unicode15.1.0/>.
+
+=head2 Assigning logical xor C<^^=> operator
+
+Perl 5.40.0 introduced the logical medium-precedence exclusive-or operator
+C<^^>.  It was not noticed at the time that the assigning variant C<^^=> was
+also missing.  This is now added.
 
 =head1 Security
 
-XXX Any security-related notices go here. In particular, any security
-vulnerabilities closed should be noted here rather than in the
-L</Selected Bug Fixes> section.
+=head2 [CVE-2024-56406] Heap buffer overflow vulnerability with tr//
 
-[ List each security issue as a =head2 entry ]
+A heap buffer overflow vulnerability was discovered in Perl.
+
+When there are non-ASCII bytes in the left-hand-side of the C<tr> operator,
+C<S_do_trans_invmap()> can overflow the destination pointer C<d>.
+
+  $ perl -e '$_ = "\x{FF}" x 1000000; tr/\xFF/\x{100}/;'
+  Segmentation fault (core dumped)
+
+It is believed that this vulnerability can enable Denial of Service or
+Arbitrary Code Execution attacks on platforms that lack sufficient defenses.
+
+Discovered by: Nathan Mills.
```

Comment:
Maybe add that the patch to fix this is applicable to all perls that are vulnerable, including those out-of-support